services/httpd/limiter.go

```diff
@@ -0,0 +1,54 @@
+package httpd
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/influxdata/influxdb/influxql"
+)
+
+// ErrConnectionLimitExceeded indicates the connection limit has been reached.
+var ErrConnectionLimitExceeded = errors.New("connection limit exceeded")
+
+// connectionLimiter limits the maximum number of connections.
+type connectionLimiter struct {
```

Comment:
If all we really care about it the underlying connection count, why not use [LimitListener](https://godoc.org/golang.org/x/net/netutil#LimitListener) at the service level? If we'd rather close the connection instead of waiting for the count to drop, we could also write something similar that uses `atomic` and does some checking to immediately accept/close connections once the limit is reached.